### DISCUSSION TITLE
variants is not working with @apply

### DISCUSSION THREAD

**Discussion by Freelancedevani** [tailwindlabs/tailwindcss#4512] [Help] [answered]
I am working with the button. when I use focus in the HTML class then it working perfectly .but when I try to use it in the  @layer component then it is generating this error. 

The `focus:` class does not exist. If `focus:` is a custom class, make sure it is defined within a `@layer` directive.

  160 |
  161 |     .vic-buttons-style {
> 162 |         @apply bg-vfc-bg w-16 h-8 bg-no-repeat text-white uppercase font-lato text-xs font-bold border-0 focus: outline-none focus:ring;
      |         ^
  163 |     }
  164 |
" -t "Laravel Mix"

html code 

`  <button id="vote" class="vic-buttons-style">Vote</button>`

css code 

@layer components {
    .vic-buttons-style {
        @apply bg-vfc-bg w-16 h-8 bg-no-repeat text-white uppercase font-lato text-xs font-bold border-0 focus: outline-none focus:ring;
    }
}

**Comment by simonswiss** [comment, marked as answer]
Hey @Freelancedevani 👋

In the code you provided, you have an empty space between `focus:` and `outline-none`, which will be interpreted as two separate classes:

![image](https://user-images.githubusercontent.com/485747/120268012-6c4b2f80-c2e8-11eb-8c12-df03419d8458.png)

Since `focus:` by itself is not a class in the generated Tailwind CSS, you get this `The focus: class does not exist.` error message!

Hope it helps 👍

**Reply by Freelancedevani** [reply]
@simonswiss thanks for your reply let me remove the space and try it again.
